### PR TITLE
(bugbash) Fixes G104 GoSec warnings

### PR DESCRIPTION
**Pull request by rajaSahil**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

- Fixes unhandled errors( G104 GoSec warnings).

### CODE REVIEW

File: pkg/verifier/duplicated_compaction.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func (DuplicatedCompactionBlocks) VerifyRepair(ctx Context, idMatcher func(ulid.
 		return errors.Errorf("id matching is not supported")
 	}
 
-	level.Info(ctx.Logger).Log("msg", "started verifying issue", "with-repair", repair)
+	_ = level.Info(ctx.Logger).Log("msg", "started verifying issue", "with-repair", repair)
```
Comment (asquare14) [inline pkg/verifier/duplicated_compaction.go:31]: I think a better option would be to do this :
```go
err = level.Info(ctx.Logger).Log("msg", "started verifying issue", "with-repair", repair)
if err!=nil {
   return err
}
```
Ignoring the error may cause issues.

**Review by asquare14 — CHANGES_REQUESTED** [review]
Same for other ```_``` statements.

File: pkg/verifier/duplicated_compaction.go (hunk 1 above)
Comment (rajaSahil) [reply pkg/verifier/duplicated_compaction.go:31]: Will it be correct to return the error?

Comment (asquare14) [reply pkg/verifier/duplicated_compaction.go:31]: yes, similar to this https://github.com/thanos-io/thanos/blob/main/pkg/verifier/duplicated_compaction.go#L34

Comment (rajaSahil) [reply pkg/verifier/duplicated_compaction.go:31]: Okay. Will update it. Thank you!

**Review by bwplotka — CHANGES_REQUESTED** [review]
Thank you for this, but I don't think it makes sense. We have tons of info log lines and we never want to check error for this particular operation (we are fine if this fails). Doing `_` for such heavily used thing is also not desired. This is why we are fine with this particular struct to not have error checked. 

That's why we have it in https://github.com/thanos-io/thanos/blob/b4d233e6fe904c52a22948afea3c3002d894e5df/.errcheck_excludes.txt#L1 (:

**Comment by yeya24** [comment]
Close this as I agree with @bwplotka

**Pull request closed without merging**